Xamarin.Forms.Core/Visual.cs

```diff
@@ -4,9 +4,20 @@ namespace Xamarin.Forms
 {
 	public static class VisualMarker
 	{
+		static bool _isMaterialRegistered = false;
+
 		public static IVisual MatchParent { get; } = new VisualRendererMarker.MatchParent();
 		public static IVisual Default { get; } = new VisualRendererMarker.Default();
 		public static IVisual Material { get; } = new VisualRendererMarker.Material();
+
+		internal static void RegisterMaterial() => _isMaterialRegistered = true;
+		internal static void MaterialCheck()
+		{
+			if (_isMaterialRegistered)
```

Comment:
How does this check work on other platforms? This looks like it will always be false on Android as there is no set from anywhere but iOS.

We could have the platform register this if it supports material...